### ISSUE TITLE
8월 1주차 회의 내용 및 heroku 배포 관련 내용입니다.

### ISSUE THREAD

**Issue by WonYong-Jang** [seokyoung-kim/TripleS-server#14]
**1. 로그인 인증 부분**

엑세스 토큰을 구글 및 네이버 프로바이더를 통해 백엔드에서 받아서 인증 진행
slack에 올려주신 node.js로 정리된 부분을 참고하여 구현 후 차주에는 프론트와 백엔드 연동 할수 있도록 준비 할 것 
register token 대신 인증 요청한 사용자가 회원가입을 한 사용자 인지 아닌지를 확인 할수 있는 값을 
같이 세션 또는 쿠키값 통해 클라이언트에 전달
토큰의 유효기간 적절하게 설정하고 jwt 입히기 

**2. 메일 발송 오픈소스 조사**

@justwaves 노드 메일러 사용해 본 경험이 있는데 해당 오픈소스가 Spring boot에도 사용할 수 있는지 확인해 봐야 할 것 같고
사용 할수 없다면 다른 오픈 소스 각자 확인 해보기 

**3. 각자 맡은 rest api 할 수 있는 부분까지 구현하여 차주에 역시 프론트와 연동 할 준비 할 것** 

**4. 어플리케이션 배포 ( heroku로 진행)**

금일 @seokyoung-kim 헤로쿠 배포하다가 에러가 나서 저도 확인을 해봤습니다. Heroku CLI를 설치하여 
진행하였고 잘 이해가 안가거나 틀린부분 의견주세요!

결론부터 말씀드리면 에러를 확인한 결과 heroku에 Chrome이 없어서 나는 에러가 한가지 있었고
mongoDB 셋업이 안되서 나는 에러가 있었습니다.

WebDriver는 gradle 라이브러리를 통해서 자동으로 셋업해주지만 실제로 Chrome 브라우저가 없기 때문에 
같이 buildpacks 부분에 추가했습니다

[https://wonyong-jang.github.io/git/2020/08/02/Github-heroku.html](https://wonyong-jang.github.io/git/2020/08/02/Github-heroku.html)

위의 링크로 정리하면서 진행하였고 mongoDB는 slack에 올려준 url을 config에 추가하였습니다. 

```
배포 URL : https://triples-app.herokuapp.com/ 이며, api/v1/cards 를 뒤에 붙여서 json 데이터를 확인 할 수 있습니다. 

Heroku git URL : https://git.heroku.com/triples-app.git 
// 위 url로 프로젝트 경로에서 clone하여 다른분들도 배포할수 있을 텐데 
// 방법을 잘 몰라서 조금더 찾아 봐야 할 것 같아요 
// @justwaves 방법을 알고 있다면 코멘트 부탁드려요  
```

기존에 테스트를 위해서 2분마다 스케줄러가 실행되게 하였는데 배포를 위해서 
임시로 오후 18시 마다 실행되도록 변경하였습니다. 확인 후 의견 부탁드립니다.

**Comment by justwaves** [comment]
카드 데이터 받아오는데 성공했습니다
프론트엔드 코드 merge 해 놨으니 테스트해보세요
카드 타입 정하는 부분 코드가 엉망이긴 한데 나중에 고쳐놓겠습니다

![screenshot](https://ifh.cc/g/Ijp4I8.jpg)

개선해야 할 부분
1. image type 추가 (이미지 비율에 따라서)
2. 브런치에서 받아오는 이미지가 저화질임
3. festa의 경우 platform 값이 null로 되어 있음

**Comment by justwaves** [comment]
heroku 협업하는 부분은 이 링크들 참조하면 될 것 같습니다

https://devcenter.heroku.com/articles/collaborating
https://devcenter.heroku.com/articles/collab

먼저 heroku에 가입된 email로 권한 주는 것부터 해야 되겠네요
제 이메일 -> justwaves@naver.com

**Comment by WonYong-Jang** [comment]
- 감사합니다 확인했습니다. 다른분들도 메일 슬랙에 있는 메일로 추가하면 될까요?

- festa의 platform 로직 추가했습니다. 

- heroku의 경우 30분간 request가 없으면 sleep 된다고 했는데 
그래서 오늘 스케줄러가 돌지 않은 것같아요. 이부분은 조금더 찾아 봐야할 것같아요.

**Comment by seokyoung-kim** [comment]
슬랙에 있는 메일로 추가해주어어어엉~~
에러 찾느라 고생햇드앙

**Comment by WonYong-Jang** [comment]
- 석형이형 추가했습니다. 아마 heroku 가입 해야할 것같은데 광진이도 heroku 가입하고 메일 남겨줘~

- client 받아서 테스트 확인했습니다. heroku에 배포한 소스는 그때 석영이형이 말한 CrossOrigin Config 소스가 
commit 안되어 있을 텐데 연동이 되네요? proxy 설정만으로 가능한걸까요?

**Comment by justwaves** [comment]
네 development 모드에서는 proxy 설정만으로 가능해요.  production 모드일 때는 proxy가 작동 안돼서 cors 설정이 필요한거예요
